Add instructions to create pages directory and add an index.js file to it (#35971)
vercel/next.js: docs/migrating/from-create-react-app.md

## Static Assets and Compiled Output

> Create React App uses the `public` directory for the [entry HTML file](https://create-react-app.dev/docs/using-the-public-folder). It is located at the project's root. Next.js uses it for static assets. When migrating from Create React App, the location of the `public` directory is the same. It's possible to add static assets in this directory, but Create React App recommends importing them directly from JavaScript files.

Reviewer: ```suggestion
Create React App uses the `public` directory for the [entry HTML file](https://create-react-app.dev/docs/using-the-public-folder) as well as static assets, but Next.js only uses it for static assets. When migrating from Create React App, the location of the `public` directory remains the same.
```

I think this can be improved

Author: The word "remains" is giving a lint-language error.

- Move any images, fonts, or other static assets to `public`.

> - Convert `index.html` (the entry point of your application) to Next.js. Any `<head>` code should be moved to a [custom `_document.js`](/docs/advanced-features/custom-document.md). Any shared layout between all pages should be moved to a [custom `_app.js`](/docs/advanced-features/custom-app.md).

Reviewer: Should this say something about converting `public/index.html` to `pages/index.js`?

Author: We do that on line `54`.

Reviewer: I think line 54 is a bit confusing because its saying "create an index.js file" but CRA already has an index file so I think renaming/converting is better?

Author: That's a good point. I'll change the instruction to say that move the `index.js` file to `pages` directory.

Reviewer: I forgot how CRA is formatted so I just did `npx create-react-app@latest` and here's what I found:

- CRA `src/index.js` seems to be similar to Next.js `pages/_document.js`
- CRA `src/App.js` seems to similar to Next.js `pages/index.js`
- CRA `public/index.html` seems to be similar to Next.js `pages/_document.js` or perhaps a layout in `pages/_app.js`

The problem I'm finding is that CRA doesn't come with a router so its almost impossible to give clear instructions here since only the user understand their own CRA application.

Author: To tackle the problem of routing, I think we do mention that Next.js has a file system based routing and provide a link for the developer to go through the link and also a link for migrating from react-router.

If we want to add conceptual information by explaining how routing works with Next.js and things they might need to go through when migrating, then we can add more information to the guide and provide basic code snippets (or not). I am not sure how Next.js team feels about adding in-depth information in this guide.

Create React App uses the `public` directory for the [entry HTML file](https://create-react-app.dev/docs/using-the-public-folder), whereas Next.js uses it for static assets. It's possible to add static assets here, but Create React App recommends importing them directly from JavaScript files.

> - Move any images, fonts, or other static assets to `public`. In a Next.js application, this directory is kept at the root of your application.

Reviewer: I don't understand this additional sentence. What does "kept" mean here?

Author: It is not, the CRA does keep at the root of the application. In the original issue for this PR, it was mentioned that the docs were not clear about where the directories should go. I thought, this would be nice to highlight for the developers who might not be familiar or just want a re-confirmation.

I can remove it if you think this is inappropriate.

Author: Removed and fixed!

> - Move any images, fonts, or other static assets to `public`. In a Next.js application, this directory is kept at the root of your application.

Reviewer 2: Maybe we can reword the above paragraph to mention that when migrating from CRA, the location of the `public` directory stays the same, at the projects root. And remove this addition:
> In a Next.js application, this directory is kept at the root of your application.

With Create React App, you're likely using React Router. Instead of using a third-party library, Next.js includes its own [file-system based routing](/docs/routing/introduction.md).

- Create a [`pages`](/docs/basic-features/pages.md) directory at the root of the application.

> - Then create a [`index.js`](https://nextjs.org/docs/routing/introduction#index-routes) file inside the `pages` directory. This file be the entry point of your Next.js application. Populate this file with code that is used to display the index route in your Create React App.

Reviewer 3: ```suggestion
- Then, create an [`index.js`](https://nextjs.org/docs/routing/introduction#index-routes) file inside the `pages` directory. This file is the entry point of your Next.js application. Populate this file with code that is used to display the index route in your Create React App.
```
